### PR TITLE
Remove redundent stats-merging logic when destroying tcache. 

### PR DESCRIPTION
**Pull request by interwq**
The stats merging is already taken care of by tcache_flush.

### CODE REVIEW

File: src/tcache.c [hunk 1]
```diff
@@ -357,25 +357,18 @@ tcache_destroy(tsd_t *tsd, tcache_t *tcache)
...
 
-		if (config_stats && tbin->tstats.nrequests != 0) {
-			arena_bin_t *bin = &arena->bins[i];
-			malloc_mutex_lock(tsd_tsdn(tsd), &bin->lock);
-			bin->stats.nrequests += tbin->tstats.nrequests;
-			malloc_mutex_unlock(tsd_tsdn(tsd), &bin->lock);
-		}
+		if (config_stats) {
+			assert(tbin->tstats.nrequests == 0);
+                }
```
Comment (jasone) [inline src/tcache.c]: This line's indentation is spaces rather than tabs, which is why the diff doesn't align as expected.

File: src/tcache.c [hunk 2]
```diff
@@ -357,25 +357,18 @@ tcache_destroy(tsd_t *tsd, tcache_t *tcache)
...
-		if (config_stats && tbin->tstats.nrequests != 0) {
-			malloc_mutex_lock(tsd_tsdn(tsd), &arena->lock);
-			arena->stats.nrequests_large += tbin->tstats.nrequests;
-			arena->stats.lstats[i - NBINS].nrequests +=
-			    tbin->tstats.nrequests;
-			malloc_mutex_unlock(tsd_tsdn(tsd), &arena->lock);
-		}
+		if (config_stats) {
+			assert(tbin->tstats.nrequests == 0);
+                }
```
Comment (jasone) [inline src/tcache.c]: Convert spaces to tabs.

File: src/tcache.c (hunk 1 above)
Comment (interwq) [inline src/tcache.c]: Ah good catch. Editor profile mix up again. Will fix.

**Comment by jasone** [comment]
Oops, I didn't notice before that the commit message needs a tweak.  The summary needs to be a single line (ideally <= 72 characters), followed by a blank line, and then further details as necessary.

**Pull request merged**